common/src/chain/transaction/partially_signed_transaction.rs

```diff
@@ -17,24 +17,35 @@ use super::{
     signature::{inputsig::InputWitness, Signable, Transactable},
     Destination, Transaction, TxOutput,
 };
-use crate::chain::{SignedTransaction, TransactionCreationError, TxInput};
+use crate::{
+    chain::{SignedTransaction, TransactionCreationError, TxInput},
+    primitives::Amount,
+};
 use serialization::{Decode, Encode};
 use utils::ensure;
 
+/// Additional info for UTXOs
+#[derive(Debug, Eq, PartialEq, Clone, Encode, Decode)]
+pub enum UtxoAdditionalInfo {
+    TokenInfo { num_decimals: u8, ticker: Vec<u8> },
+    PoolInfo { staker_balance: Amount },
+    NoAdditionalInfo,
+}
+
 #[derive(Debug, Eq, PartialEq, Clone, Encode, Decode)]
 pub struct PartiallySignedTransaction {
     tx: Transaction,
     witnesses: Vec<Option<InputWitness>>,
 
-    input_utxos: Vec<Option<TxOutput>>,
+    input_utxos: Vec<Option<(TxOutput, UtxoAdditionalInfo)>>,
     destinations: Vec<Option<Destination>>,
 }
 
 impl PartiallySignedTransaction {
     pub fn new(
         tx: Transaction,
         witnesses: Vec<Option<InputWitness>>,
-        input_utxos: Vec<Option<TxOutput>>,
+        input_utxos: Vec<Option<(TxOutput, UtxoAdditionalInfo)>>,
         destinations: Vec<Option<Destination>>,
     ) -> Result<Self, TransactionCreationError> {
```

Comment:
This is indeed rather hacky. Can we put this info somewhere beside `PartiallySignedTransaction` instead?

Reply:
I agree that the way it's done right now is hacky. 
Maybe moving `PartiallySignedTransaction` from `wallet` to `common` was not that good idea after all. I did that few months ago because the struct looked really basic and a good companion for `SignedTransaction`,  also I thought I'd use it in tx-verifier. But maybe I misunderstood it and the type is from wallet layer. Thoughts?

Reply:
I agree that the way it's done right now is hacky.
Maybe moving `PartiallySignedTransaction` from `wallet` to `common` was not that good idea after all. I did that few months ago because the struct looked really basic and a good companion for `SignedTransaction`, also I thought I'd use it in `tx-verifier`. But maybe I misunderstood it and the type is from wallet layer. Thoughts?